git-open

```diff
@@ -42,8 +42,9 @@ if ! git rev-parse --is-inside-work-tree &>/dev/null; then
 fi
 
 # choose remote. priority to: provided argument, default in config, detected tracked remote, 'origin'
-branch_name=$(git name-rev --name-only HEAD 2>/dev/null)
-tracked_remote=$(git config "branch.$branch_name.remote")
+branch=${2:-$(git symbolic-ref -q --short HEAD)}
+upstream_branch=$(git config "branch.$branch.merge" | sed "s/^refs\/heads\///")
```

Comment:
https://git-scm.com/docs/git-config#Documentation/git-config.txt-branchltnamegtmerge It looks like this could have multiple values but git config will just get the first value so it should be fine.

I think we can convert a lot of these lines to `git for-each-ref`. 

`git for-each-ref --format='%(upstream:short)' "$(git symbolic-ref -q HEAD)"`. will return `origin/upstream-branch` for example and then it can be split like the urlpath is below. I feel like we may be able to simplify some of the existing code here

Reply:
I'm fine with pushing this off, it's not a big deal. I would like to not use sed though, for now something like this would work:

```bash
branch_ref="$(git symbolic-ref -q HEAD)"
tracked_remote="$(git for-each-ref --format='%(upstream:remotename)' "$branch_ref")"
upstream_branch="$(git for-each-ref --format='%(upstream:lstrip=-1)' "$branch_ref")"
```

Reply:
Travis tests fails for `remotename`, is it new? `fatal: unrecognized %(upstream:remotename) argument: remotename`

Other than that I wouldn't mind `for-each-ref`, though it would need some more tweaking to handle user input.

Also `lstrip=-1` wouldn't work for me since I have a `/` in my remote branch name, though `lstrip=3` seems to to do the trick

Reply:
Did another attempt without `sed` using [parameter-substitution](http://www.tldp.org/LDP/abs/html/parameter-substitution.html)

```bash
upstream_branch_full_name=$(git config "branch.$branch.merge")
upstream_branch=${upstream_branch_full_name#"refs/heads/"}
```

Reply:
Good call looks semi-new (from 2017): https://github.com/git/git/commit/cc72385fe350d4d7bc64f84e8817dbdfe27b04fe.